rosette/src/BaseSupp.cc

```diff
@@ -521,6 +523,8 @@ DEF("char*->string", char_star_to_string, 1, 1) {
         return RBLstring::create((char*)addr);
     else
         PRIM_ERROR("invalid address");
+
+    return FIXNUM(0);
```

Comment:
Isn't this dead code?

Reply:
It looks like PRIM_ERROR ends up calling the object's runtimeError method, which in turn returns "DEADTHREAD" which is a pOb that is built up from data in the object.

In some places, functions defined with "DEF" return the result of PRIM_ERROR, and in others nothing is returned.

I guess I could modify these cases to return the result of PRIM_ERROR.  I just don't know why this wasn't done in the first place.

Reply:
@kjmcqueen can you point me to a primitive where nothing is returned?

Reply:
@tymm, unless I'm missing something, this exact spot DEF("char*->string", ... is an example. The code either calls RBLstring::create and returns the result, or the macro PRIM_ERROR calls the primitive's runtimeError method which returns "DEADTHREAD", but that return value is dropped, and nothing is returned from char_star_to_string.

I admit, I haven't quite gotten the hang of unraveling these multi-level nested pre-processor macros.

Reply:
FWIW, all of these places where I've added "return FIXNUM(0);" are examples of places where the compiler warned that nothing was being returned from a function that was declared as returning a value.

I'm just trying to make the compiler happy because if a calling function is expecting a return value, and there is none, it will get whatever random garbage happens to be in the uninitialized return value.

Reply:
**Final:** We will use `return PRIM_ERROR(...)`.